connectors/vault/vaultconn-bl/vault_reader_test.go

```diff
@@ -0,0 +1,92 @@
+package vaultconnbl
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/vault"
+
+	tu "github.com/ibm/the-mesh-for-data/connectors/vault/testutil"
+	vltutils "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+)
+
+// TestVaultConnectorNormalRun tests the execution of GetCredentialsInfo with dummy mock id and
+// dummy credentials stored in the runtime test instance of vault
+func TestVaultConnectorNormalRun(t *testing.T) {
+	ln, client := createTestVault(t)
+	defer ln.Close()
+
+	srv := &Server{}
+	appID := "mock-appID"
+	//datasetID := "{\"ServerName\":\"cocoMDS3\",\"AssetGuid\":\"f6d9bf8c-dd37-4747-bca9-8ca9d1a5bb8f\"}"
+	datasetID := "mock-datasetID"
+	objToSendForCredential := &pb.DatasetCredentialsRequest{AppId: appID, DatasetId: datasetID}
+
+	config := vltutils.VaultConfig{
+		Token:   "token",
+		Address: "address",
+	}
+	connection := vltutils.VaultConnection{
+		Config: config,
+	}
+
+	data := map[string]interface{}{
+		"dataset_id":  datasetID,
+		"credentials": "my_egeria_credentials_test",
+	}
+
+	_, err := client.Logical().Write("secret/"+datasetID, data)
+	if err != nil {
+		t.Errorf("Error writing credentials from vault for " + datasetID + ":" + err.Error())
+	}
+
+	connection.Client = client
+	userVaultPath := tu.GetEnvironment()
+	var vault vltutils.VaultConnection
+	vault = connection
+	credentialsInfo, _ := srv.GetCredentialsInfo(objToSendForCredential, vault, userVaultPath)
+	log.Println("credentialsInfo")
+	log.Println(credentialsInfo)
+
+	expectedCredentials := tu.GetExpectedVaultCredentials(objToSendForCredential)
+	tu.EnsureDeepEqualCredentials(t, credentialsInfo, expectedCredentials)
+}
+
+func createTestVault(t *testing.T) (net.Listener, *api.Client) {
+	t.Helper()
+
+	// Create an in-memory, unsealed core (the "backend", if you will).
+	core, keyShares, rootToken := vault.TestCoreUnsealed(t)
+	_ = keyShares
+
+	// Start an HTTP server for the core.
+	ln, addr := http.TestServer(t, core)
```

Comment:
Why do we need an http server if we run the vault tests against an in-memory vault client?

Reply:
This is starting in memory vault server per unit test to avoid writing brittle unit tests. It is a public api exposed by vault  which is useful for making the unit test very self contained. You can check the explanation here in this video from 53.58 to  57.51 seconds (https://www.youtube.com/watch?v=yszygk1cpEc&feature=youtu.be&t=3238).